### PR TITLE
[MM-55497] Add tooltips with enlarged emojis when hovering over emojis within message content

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

This PR adds the tooltip in emojis on hover at post messages.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/26519

#### Release Note

```release-note
Added emoji tooltips on hover in post message.
```

### CODE REVIEW

File: webapp/channels/src/components/post_emoji/post_emoji.tsx [hunk 1]
```diff
@@ -22,14 +24,22 @@ const PostEmoji = ({name, imageUrl}: Props) => {
...
+            emoji={name}
+            emojiStyle='large'
+            placement='top'
         >
-            {emojiText}
-        </span>
+            <span
+                alt={emojiText}
+                className='emoticon'
+                data-testid={`postEmoji.${emojiText}`}
```
Comment (larkox) [inline webapp/channels/src/components/post_emoji/post_emoji.tsx:37]: Why the change on the test id?

File: webapp/channels/src/components/post_emoji/post_emoji.tsx [hunk 2]
```diff
@@ -22,14 +24,22 @@ const PostEmoji = ({name, imageUrl}: Props) => {
     }
 
     return (
-        <span
-            alt={emojiText}
-            className='emoticon'
+        <WithTooltip
```
Comment (larkox) [inline webapp/channels/src/components/post_emoji/post_emoji.tsx:27]: The way this works right now, I worry what happens if there are many emojis in screen. Could you do a quick test for me?

See if there is any visible difference between this PR and master when you have a post with many emojis (copy paste `:+1: ` until you get to the maximum number of characters).

The issue I see is that `WithTooltip` creates a div like this every time it is present:

`<div style="display: flex; flex-direction: column; align-items: center; z-index: 1070; position: absolute; top: -1000px; left: -1000px;" class="tooltip-container"></div>` and if we are rendering a lot of emojis, we may create many of these divs, and I don't know how troublesome for the browser that is.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! I mark it as request changes because I want you to do a test before I am happy with the merge. If the test goes fine, I will approve. If not, we will reconsider how to handle this.

File: webapp/channels/src/components/post_emoji/post_emoji.tsx (hunk 1 above)
Comment (tanmaythole) [reply webapp/channels/src/components/post_emoji/post_emoji.tsx:37]: As `title` attribute in the span tag renders the default tooltip which is not required in this case. `testid` added here to find the required element in unit tests.

File: webapp/channels/src/components/post_emoji/post_emoji.tsx (hunk 2 above)
Comment (tanmaythole) [reply webapp/channels/src/components/post_emoji/post_emoji.tsx:27]: This div will created when hover is active for the emoji. Once hover exited for the element the created div is destroyed from the document.

**Review by larkox — APPROVED** [review]

File: webapp/channels/src/components/post_emoji/post_emoji.tsx [hunk 3]
```diff
@@ -22,14 +24,22 @@ const PostEmoji = ({name, imageUrl}: Props) => {
...
+            id='postEmoji__tooltip'
```
Comment (hmhealey) [inline webapp/channels/src/components/post_emoji/post_emoji.tsx:28]: @larkox I'm not sure if you're aware of this, but this ID prop doesn't actually need to be unique with how we're using React Bootstrap for it. I thought that this needed to be unique since it's passed through to the DOM, but since only one `WithTooltip` ever renders its tooltip at a time, it doesn't end up needing to be unique

**Review by hmhealey — APPROVED** [review]
Thanks for the PR, @tanmaythole! This looks good to me!

I left a comment below about something I was going to ask about, but it's about the `WithTooltip` and not something to be changed in this PR

**Review by yasserfaraazkhan — APPROVED** [review]
